### PR TITLE
Framework.yaml export

### PR DESCRIPTION
**Pull request by stinkyfingers**
# Description

https://shipaio.atlassian.net/secure/RapidBoard.jspa?rapidView=1&projectKey=SHIPA&modal=detail&selectedIssue=SHIPA-1547

Adds `framework export` command to export the specified framework spec to a yaml file. 

Fixes # 1547

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Chore (documentation addition or typo, file relocation)

## Testing

- [x] New tests were added with this PR that prove my fix is effective or that my feature works (describe below this bullet)
- [ ] This change requires no testing (i.e. documentation update)

## Documentation

- [x] All added public packages, funcs, and types have been documented with doc comments
- [x] I have commented my code, particularly in hard-to-understand areas

## Final Checklist:

- [x] I followed standard [GitHub flow](https://guides.github.com/introduction/flow/) guidelines
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings

## Additional Information (omit if empty)

Please include anything else relevant to this PR that may be useful to know.

### CODE REVIEW

File: internal/api/v1beta1/framework_types.go [hunk 1]
```diff
@@ -53,12 +53,14 @@ type FrameworkList struct {
 
 // FrameworkSpec defines the desired state of Framework
 type FrameworkSpec struct {
+	Version string `json:"version,omitempty" yaml:"version,omitempty"`
+	Name    string `json:"-" yaml:"name"`
 	// +kubebuilder:validation:MinLength=1
-	NamespaceName string `json:"namespace"`
+	NamespaceName string `json:"namespace" yaml:"namespace"`
```
Comment (aleksej-paschenko) [inline internal/api/v1beta1/framework_types.go:59]: I think we can use `sigs.k8s.io/yaml` and drop `yaml` tags. The reason for this is that `sigs.k8s.io/yaml` uses json tags and this means yaml and json representations are always the same

**Review by aleksej-paschenko — CHANGES_REQUESTED** [review]
looks awesome, let's just change yaml package and drop `yaml` tags

File: internal/api/v1beta1/framework_types.go (hunk 1 above)
Comment (stinkyfingers) [reply internal/api/v1beta1/framework_types.go]: I made the switch to sigs.k8.io/yaml. My only concern was that this lib alphabetizes fields, so you end up with fields  like `name` and `version` near the bottom of the yaml output.

File: cmd/ketch/framework_export_test.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	ketchv1 "github.com/shipa-corp/ketch/internal/api/v1beta1"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/shipa-corp/ketch/internal/mocks"
+	"k8s.io/apimachinery/pkg/runtime"
```
[the remark this answers is not in the document]
Comment (stinkyfingers) [reply cmd/ketch/framework_export_test.go]: Done.

**Review by aleksej-paschenko — APPROVED** [review]
looks awesome

File: internal/api/v1beta1/framework_types.go (hunk 1 above)
Comment (aleksej-paschenko) [reply internal/api/v1beta1/framework_types.go]: oops, that's a valid concern!

**Pull request merged**